NewsApi/analyzer.py

```diff
@@ -0,0 +1,111 @@
+"""
+    This contains sentiment analysis models
+"""
+import json
+import pandas as pd
+import nltk
+nltk.download('vader_lexicon')
+
+from nltk.sentiment.vader import SentimentIntensityAnalyzer
+from flask import request
+from newsapi import NewsApiClient
+from NewsApi.api_key import api_key
+from sqlalchemy import null, exc
+
+
+class DataProcessor:
```

Comment:
The fetching of data and sentiment analysis can be two different modules (files).
In Python, files are used to group together related functions and classes are mainly used for representing Entities in your code eg a User, or a News article.